src/Geosuggest.jsx

```diff
@@ -414,58 +438,66 @@ class Geosuggest extends React.Component {
    */
   render() {
     const attributes = filterInputAttributes(this.props),
-      classes = classnames(
-        'geosuggest',
-        this.props.className,
-        {'geosuggest--loading': this.state.isLoading}
-      ),
+      classes = classnames('geosuggest', this.props.className, {
+        'geosuggest--loading': this.state.isLoading
+      }),
       shouldRenderLabel = this.props.label && attributes.id,
-      input = <Input className={this.props.inputClassName}
-        ref={i => this.input = i}
-        value={this.state.userInput}
-        doNotSubmitOnEnter={!this.state.isSuggestsHidden}
-        ignoreTab={this.props.ignoreTab}
-        ignoreEnter={this.props.ignoreEnter}
-        style={this.props.style.input}
-        onChange={this.onInputChange}
-        onFocus={this.onInputFocus}
-        onBlur={this.onInputBlur}
-        onKeyDown={this.props.onKeyDown}
-        onKeyPress={this.props.onKeyPress}
-        onNext={this.onNext}
-        onPrev={this.onPrev}
-        onSelect={this.onSelect}
-        onEscape={this.hideSuggests} {...attributes} />,
-      suggestionsList = <SuggestList isHidden={this.state.isSuggestsHidden}
-        style={this.props.style.suggests}
-        suggestItemStyle={this.props.style.suggestItem}
-        userInput={this.state.userInput}
-        isHighlightMatch={this.props.highlightMatch}
-        suggestsClassName={this.props.suggestsClassName}
-        suggestItemClassName={this.props.suggestItemClassName}
-        suggests={this.state.suggests}
-        hiddenClassName={this.props.suggestsHiddenClassName}
-        suggestItemActiveClassName={this.props.suggestItemActiveClassName}
-        activeSuggest={this.state.activeSuggest}
-        onSuggestNoResults={this.onSuggestNoResults}
-        onSuggestMouseDown={this.onSuggestMouseDown}
-        onSuggestMouseOut={this.onSuggestMouseOut}
-        onSuggestSelect={this.selectSuggest}
-        renderSuggestItem={this.props.renderSuggestItem}
-        minLength={this.props.minLength}/>;
-
-    return <div className={classes}>
-      <div className="geosuggest__input-wrapper">
-        {shouldRenderLabel &&
-          <label className="geosuggest__label"
-            htmlFor={attributes.id}>{this.props.label}</label>
-        }
-        {input}
-      </div>
-      <div className="geosuggest__suggests-wrapper">
-        {suggestionsList}
+      input = (
+        <Input
+          className={this.props.inputClassName}
+          ref={i => (this.input = i)}
+          value={this.state.userInput}
+          doNotSubmitOnEnter={!this.state.isSuggestsHidden}
+          ignoreTab={this.props.ignoreTab}
+          ignoreEnter={this.props.ignoreEnter}
+          style={this.props.style.input}
+          onChange={this.onInputChange}
+          onFocus={this.onInputFocus}
+          onBlur={this.onInputBlur}
+          onKeyDown={this.props.onKeyDown}
```

Comment:
How about deconstructing the `this.props` and `this.state` at the beginning of `render`? Makes for better readability in the components.

Reply:
This would be a refactoring, not just a tooling upgrade.